### DISCUSSION TITLE
Get single Node feature instead of global pooling

### DISCUSSION THREAD

**Discussion by oliverluo1993** [pyg-team/pytorch_geometric#4080] [Q&A] [answered]
Hi,

I am trying to predict the node level instead of graph level properties.

Except from global pooling layers to get graph level information. Is there a way to fetch a single node feature from batched node feature matrix? 

Best regards

**Comment by rusty1s** [comment, marked as answer]
You can use traditional indexing/`index_select` to select specific node feature vectors, e.g.
```python
node_idx = torch.tensor([0, 10, 15])  # Select features of node 0, 10 and 15
out = x[node_idx]
```
Does that answer your question?

**Reply by oliverluo1993** [reply]
Thanks! 

But in this way, it only works with batch size = 1. 

Is there a way to work with mini-batching of pytorch geometric?

**Reply by rusty1s** [reply]
If you name your tensor to include the word `index`, mini-batching should increase indices out-of-the-box:
```python
data.node_index = ...
```

**Reply by oliverluo1993** [reply]
Thanks! I will try that!
